server/aesdsocket.c

```diff
@@ -0,0 +1,565 @@
+/***********************************************************************
+ * @file      		aesdsocket.c
+ * @version   		0.1
+ * @brief		Socket server application code
+ *
+ * @author    		Amey More, Amey.More@Colorado.edu
+ * @date      		Oct 4, 2023
+ *
+ * @institution 	University of Colorado Boulder (UCB)
+ * @course      	ECEN 5713: Advanced Embedded Software Development
+ * @instructor  	Dan Walkes
+ *
+ * @assignment 	Assignment 5
+ * @due        	Oct 08, 2023 at 11:59 PM
+ *
+ * @references
+ * https://www.gnu.org/software/libc/manual/html_node/
+ * Termination-in-Handler.html
+ *
+ * https://www.thegeekstuff.com/2012/02/c-daemon-process/ 
+ *
+ * https://beej.us/guide/bgnet/html/#getaddrinfoprepare-to-launch
+ ************************************************************************/
+#include "aesdsocket.h"
+
+volatile sig_atomic_t cleanup = 0;
+
+// For socket commands
+command_status_t command_status;
+bool deamon_mode = false;
+
+// Outout data file
+char *data_file = "/var/tmp/aesdsocketdata";
+int data_file_fd;
+
+// Server & Client Socket
+int socket_fd;
+int accept_fd;
+
+// prototypes
+void global_clean_up();
+void socket_application();
+
+// https://www.gnu.org/software/libc/manual/html_node/
+// Termination-in-Handler.html
+void handle_termination(int signo)
+{
+	if(signo == SIGINT || signo == SIGTERM)
+	{
+		syslog(LOG_INFO,"Caught signal, exiting\n");
+
+		/* 
+		* Since this handler is established for more than one kind of signal, 
+		* it might still get invoked recursively by delivery of some other 
+		* kind of signal. Use a static variable to keep track of that.
+		*/
+		if (cleanup)
+			raise (signo);
+		cleanup = 1;
+		
+		/*
+		* Clean-up actions
+		*/
+		global_clean_up();		
+
+		/* 
+		* Now reraise the signal.  We reactivate the signal’s
+		* default handling, which is to terminate the process.
+		* We could just call exit or abort,
+		* but reraising the signal sets the return status
+		* from the process correctly.
+		*/
+		signal(signo, SIG_DFL);
+		raise(signo);
```

Comment:
@ameyflash consider a design where you can handle cleanup in your main loop instead of exiting directly from your signal handler.  Correct cleanup will be more difficult to do as the design gets more complicated in assignment 6.